### DISCUSSION TITLE
How to integrate a custom panel directly in the code base

### DISCUSSION THREAD

**Discussion by charlielito** [foxglove/community#946] [Q&A]
According to https://foxglove.dev/blog/building-a-custom-react-panel-with-foxglove-studio-extensions we can create a custom panel but we need to zip it and publish it as an extension in the marketplace. I was wondering how we can leave the custom panel in a fork without the need to publish it

**Comment by defunctzombie** [comment]
We do not support that workflow. Extension installation and management is only supported through publishing on the marketplace or your foxglove organization account.

https://docs.foxglove.dev/docs/visualization/extensions/publish/#sharing-with-your-team
